src/tools/cephfs_mirror/PeerReplayer.cc

```diff
@@ -1334,6 +1334,7 @@ int PeerReplayer::do_synchronize(const std::string &dir_root, const Snapshot &cu
         sync_stack.emplace(SyncEntry(epath, stx));
       }
     } else {
+      std::scoped_lock locker(m_lock);
```

Comment:
This is incorrect - snap sync happens outside the lock to allow concurrent snap syncs (to avoid holding the lock for long).

Reply:
right, it doesn't work too.

Reply:
@vshankar Dropped it.